sympy/solvers/ode.py

```diff
@@ -4003,6 +4003,40 @@ def _frobenius(n, m, p0, q0, p, q, x0, x, c, check=None):
 
     return frobdict
 
+
+def _check_substitution_type(eq, func):
+    r"""
+    Matches any differential equation that order_reducing_substitution can solve.
+
+    For this to work equation should min order of derivative to be 1, i.e., f(x) should not be present.
+    """
+
+    x = func.args[0]
+    f = func.func
+    order_to_subs = 0
+    D = Dummy()
+    for nsubs_try in range(1,ode_order(eq,f(x))):
+        if eq.subs(f(x).diff(x, nsubs_try), D).has(f(x)):
+            break
+        else:
+            order_to_subs += 1
+    if order_to_subs > 0:
+        return {'var':order_to_subs, 'solutions':True}
```

Comment:
I would rather explicitly `return None` if the method doesn't match

Reply:
Won't this end up being the gcd of the orders? So if `gcd(*[d.variable_count[0][1] for d in eq.atoms(Derivative) if len(d.variable_count) == 1 and d.variable_count[0][0] == x]) == 1` (or fails because there is only one Derivative present) then there is no substitution, otherwise there is?

Reply:
Rather, won't this just be the minimum order present? But only if it is the gcd of the other ordered. What does your routine do with `Derivative(f(x),x,3) + Derivative(f(x),x,2)`?

Reply:
I would expect it to write it as .
g' + g = 0
g = f''
then integrate g two times.

Reply:
It will be the minimum order derivative of `f(x)`. It would be good to reduce the cost of checking in the case where this doesn't match I think.

Most of the time we can decide that the method does not match by testing if any f(x) appears outside of a Derivative. I can picture how to do that when looking at an expression tree but not sure how to turn that into code.

Reply:
`func_outside = expr.xreplace(dict([(d, Dummy()) for d in expr.atoms(Derivative)])).has(f(x))`

Reply:
If `func_outside` is True the the method won't work; if it is false then it might not work if there is a case like trying to solve `f(x).diff(x)` and having `Derivative(y*f(x), y)` present. (Hmm...there is that pesky constant-in-object case again.) So in that case, your loop will still be done to double check.

Reply:
Well there is a break statement which will abort the loop as soon as it finds a derivative of order 0.

Reply:
Otherwise too the loop will break in a single step if f(x) is found.

Reply:
actually loop is not required since minimum order derivative is being used and substituting it would result in it being 0 order derivative.

Reply:
Tricky case would be something like `Derivative(f(x)*f(x).diff(x), x)` which will have an undifferentiated `f(x)` once expanded.

Reply:
> I would rather explicitly `return None` if the method doesn't match

I agree. Can you just return None or the order?
```python
def _check_substitution_type_match(eq, func):
    r"""
    Matches any differential equation that `order_reducing_substitution` can solve.
    For this to work equation should have a minimum order of derivative to be 1, i.e.,
    `f(x)` should not be present.
    """
    if len(func.args) != 1:
        return
    x = func.args[0]
    D = Dummy()
    for order_to_subs in range(1, ode_order(eq, func)):
        if eq.subs(func.diff(x, order_to_subs), D).has(func):
            return
    return order_to_subs
```

Also

Reply:
Why is the order needed to be returned if it is not going to the "hints" list?